dogecoin/params.go

```diff
@@ -0,0 +1,258 @@
+	TargetTimePerBlock:       time.Minute, // 1 minute
+	RetargetAdjustmentFactor: 4,           // 25% less, 400% more
+	ReduceMinDifficulty:      false,
+	MinDiffReductionTime:     0,
+	GenerateSupported:        false,
+
+	// Checkpoints ordered from oldest to newest.
+	Checkpoints: []chaincfg.Checkpoint{
+		{Height: 0, Hash: newHashFromStr("bb0a78264637406b6360aad926284d544d7049f45189db5664f3c4d07350559e")},
+		{Height: 483173, Hash: newHashFromStr("a804201ca0aceb7e937ef7a3c613a9b7589245b10cc095148c4ce4965b0b73b5")},
+		{Height: 591117, Hash: newHashFromStr("5f6b93b2c28cedf32467d900369b8be6700f0649388a7dbfd3ebd4a01b1ffad8")},
+		{Height: 658924, Hash: newHashFromStr("ed6c8324d9a77195ee080f225a0fca6346495e08ded99bcda47a8eea5a8a620b")},
+		{Height: 703635, Hash: newHashFromStr("839fa54617adcd582d53030a37455c14a87a806f6615aa8213f13e196230ff7f")},
+		{Height: 1000000, Hash: newHashFromStr("1fe4d44ea4d1edb031f52f0d7c635db8190dc871a190654c41d2450086b8ef0e")},
+		{Height: 1202214, Hash: newHashFromStr("a2179767a87ee4e95944703976fee63578ec04fa3ac2fc1c9c2c83587d096977")},
+		{Height: 1250000, Hash: newHashFromStr("b46affb421872ca8efa30366b09694e2f9bf077f7258213be14adb05a9f41883")},
+		{Height: 1500000, Hash: newHashFromStr("0caa041b47b4d18a4f44bdc05cef1a96d5196ce7b2e32ad3e4eb9ba505144917")},
+		{Height: 1750000, Hash: newHashFromStr("8042462366d854ad39b8b95ed2ca12e89a526ceee5a90042d55ebb24d5aab7e9")},
+		{Height: 2000000, Hash: newHashFromStr("d6acde73e1b42fc17f29dcc76f63946d378ae1bd4eafab44d801a25be784103c")},
+		{Height: 2250000, Hash: newHashFromStr("c4342ae6d9a522a02e5607411df1b00e9329563ef844a758d762d601d42c86dc")},
+		{Height: 2500000, Hash: newHashFromStr("3a66ec4933fbb348c9b1889aaf2f732fe429fd9a8f74fee6895eae061ac897e2")},
+		{Height: 2750000, Hash: newHashFromStr("473ea9f625d59f534ffcc9738ffc58f7b7b1e0e993078614f5484a9505885563")},
+		{Height: 3062910, Hash: newHashFromStr("113c41c00934f940a41f99d18b2ad9aefd183a4b7fe80527e1e6c12779bd0246")},
+	},
+
+	// Consensus rule change deployments.
+	//
+	// The miner confirmation window is defined as:
+	//   target proof of work timespan / target proof of work spacing
+	RuleChangeActivationThreshold: 9576,  // 95% of MinerConfirmationWindow
+	MinerConfirmationWindow:       10080, //
+	Deployments: [chaincfg.DefinedDeployments]chaincfg.ConsensusDeployment{
+		chaincfg.DeploymentTestDummy: {
+			BitNumber:  28,
+			StartTime:  1199145601, // January 1, 2008 UTC
+			ExpireTime: 1230767999, // December 31, 2008 UTC
+		},
+		chaincfg.DeploymentCSV: {
+			BitNumber:  0,
+			StartTime:  1462060800, // May 1, 2016 UTC
+			ExpireTime: 1493596800, // May 1, 2017 UTC
+		},
+		chaincfg.DeploymentSegwit: {
+			BitNumber:  1,
+			StartTime:  1479168000, // November 15, 2016 UTC
+			ExpireTime: 0,          // Disabled
+		},
+	},
+
+	// Mempool parameters
+	RelayNonStdTxs: false,
+
+	// Human-readable part for Bech32 encoded segwit addresses, as defined in
+	// BIP 173.
+	Bech32HRPSegwit: "bc", // always bc for main net
```

Comment:
Although currently unimplemented, this is planned to be "doge" in 0.21, see [chainparams.cpp in 8b0dcb5 on dogecoin/dogecoin](https://github.com/dogecoin/dogecoin/blob/8b0dcb523dcda768988f0ed661a70611c3d727b3/src/chainparams.cpp#L137)